addon/core/handler.ts

```diff
@@ -161,8 +161,10 @@ export default class TableHandler {
       .upsertColumns({
         columns: this.columns.filter((column) => column.definition.key !== definition.key)
       })
-      .then((resp) => {
-        this.columns = resp.columns;
+      .then(({ columns }) => {
+        this.columns = columns;
+        this._reinitColumnsAndRows(columns);
```

Comment:
maybe you can do this only if the column removed (you can look it up via the definition) had filters/orders ? this way we can avoid doing a useless api call